server/CGameHandler.cpp

```diff
@@ -999,18 +999,21 @@ void CGameHandler::run(bool resume)
 	turnOrder->onGameStarted();
 
 	//wait till game is done
+	auto clockLast = std::chrono::high_resolution_clock::now();
 	while(lobby->getState() == EServerState::GAMEPLAY)
 	{
-		const int waitTime = 100; //ms
+		const auto clockDuration = std::chrono::high_resolution_clock::now() - clockLast;
+		const int timePassed = std::chrono::duration_cast<std::chrono::milliseconds>(clockDuration).count();
+		clockLast += clockDuration;
 
 		for(PlayerColor player(0); player < PlayerColor::PLAYER_LIMIT; ++player)
 			if(gs->isPlayerMakingTurn(player))
-				turnTimerHandler.onPlayerMakingTurn(player, waitTime);
+				turnTimerHandler.onPlayerMakingTurn(player, timePassed);
 
 		if(gs->curB)
-			turnTimerHandler.onBattleLoop(waitTime);
+			turnTimerHandler.onBattleLoop(timePassed);
```

Comment:
Mutex should probably be locked in this method, not inside turnTimeHandler.
Or you need to remove all checks from here - for example, I am not sure if gs->isPlayerMakingTurn can be accessed from multiple threads

Reply:
Or perhaps - move all timer-related code from here (including all checks) into method inside turnTimeHandler, e.g. `TurnTimerHandler::update()`?